src/main/java/org/folio/circulation/domain/notice/schedule/DueDateScheduledNoticeHandler.java

```diff
@@ -92,13 +105,26 @@ public DueDateScheduledNoticeHandler(
       .thenApply(r -> r.mapFailure(this::handleFailure));
   }
 
+  private Result<Response> failIfTemplateNotFound(Response response, String templateId) {
+    if (response.getStatusCode() == 404) {
```

Comment:
Can the duplication between this and the method in DueDateNotRealTimeScheduledNoticeHandler.class be removed? Possibly by moving out the method to util class.

Reply:
I didn't find appropriate util class for it and decided to not create new one. Do you think it makes sense?

Reply:
I agree that this duplication could be removed. Maybe the repository is an appropriate place for this logic (or part of it)?

Reply:
moved to the repository